auth/src/main/java/com/firebase/ui/auth/ui/phone/VerifyPhoneNumberFragment.java

```diff
@@ -273,6 +272,23 @@ private void setCountryCode(PhoneNumber phoneNumber) {
         }
     }
 
+    private void setupPrivacyDisclosures(TextView footerText) {
+        final String verifyPhoneButtonText = getString(R.string.fui_verify_phone_number);
+        final String multipleProviderFlowText = getString(R.string.fui_sms_terms_of_service,
+                verifyPhoneButtonText);
+        FlowParameters flowParameters = getFlowParams();
+        PrivacyDisclosureUtils.setupTermsOfServiceAndPrivacyPolicySmsText(getContext(),
+                flowParameters,
+                mSmsTermsText,
+                multipleProviderFlowText);
+
+        if (!flowParameters.isSingleProviderFlow()) {
```

Comment:
So if this **is** a single provider flow then we'll have both the big text and the footer?  If that's not the case (maybe the previous method has some internal switching) let's make that clearer here by doing `if / else` so only one of the `PrivacyDisclosureUtils` classes runs.

Reply:
Yes, SMS is special because we warn them either way (about the SMS cost).